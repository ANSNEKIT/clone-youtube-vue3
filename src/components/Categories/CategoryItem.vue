<template>
    <li>
        <button
            class="px-3 py-1 h-[32px] min-w-[50px] max-w-[15.5rem] text-base truncate cursor-pointer border rounded-full bg-black/5 hover:bg-black/10 focus:bg-black/10"
            :class="{ '!bg-black !text-white': isActive }"
            @click="$emit('changeActiveCategory', id)"
        >
            {{ category }}
        </button>
    </li>
</template>

<script setup lang="ts">
defineProps({
    id: {
        type: Number,
        required: true,
    },
    isActive: {
        type: Boolean,
        default: false,
    },
    category: {
        type: String,
        default: '',
    },
})
defineEmits(['changeActiveCategory'])
</script>
