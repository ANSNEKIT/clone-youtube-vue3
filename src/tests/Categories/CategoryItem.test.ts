import { render, screen } from '@testing-library/vue'
import CategoryItem from '@/components/Categories/CategoryItem.vue'

const category = 'One very very long longest category'

function renderCategoryItem() {
    const options = {
        props: {
            id: 0,
            category,
        },
    }

    return render(CategoryItem, options)
}

it('render with label and not active', () => {
    renderCategoryItem()

    expect(screen.getByText(category)).not.toHaveClass('!bg-black !text-white')
})

it('render correctly', () => {
    const { html } = renderCategoryItem()

    expect(html()).toMatchSnapshot()
})
