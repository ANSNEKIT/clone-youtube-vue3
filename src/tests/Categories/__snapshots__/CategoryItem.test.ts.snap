// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `"<li><button class="px-3 py-1 h-[32px] min-w-[50px] max-w-[15.5rem] text-base truncate cursor-pointer border rounded-full bg-black/5 hover:bg-black/10 focus:bg-black/10">One very very long longest category</button></li>"`;
